java/src/test/java/com/mlt/TestUtils.java

```diff
@@ -3,123 +3,95 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import com.mlt.converter.mvt.MapboxVectorTile;
+import com.mlt.data.Feature;
 import com.mlt.data.MapLibreTile;
 import com.mlt.vector.FeatureTable;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
 
 public class TestUtils {
+
+  private static int compareFeatures(
+      List<Feature> mltFeatures, List<Feature> mvtFeatures, boolean isFeatureTableSorted) {
+    int numErrors = 0;
+    for (var j = 0; j < mltFeatures.size(); j++) {
+      var mltFeature = mltFeatures.get(j);
+      var mvtFeature =
+          isFeatureTableSorted
+              ? mvtFeatures.stream().filter(f -> f.id() == mltFeature.id()).findFirst().get()
+              : mvtFeatures.get(j);
+      assertEquals(mvtFeature.id(), mltFeature.id());
+      var mltGeometry = mltFeature.geometry();
+      var mvtGeometry = mvtFeature.geometry();
+      if (!mvtGeometry.equalsExact(mltGeometry)) {
+        // System.out.println("Failure comparing geometry for feature: " + mvtFeature.id());
+        // System.out.println("    mvtGeometry: " + mvtGeometry);
+        // System.out.println("    mltGeometry: " + mltGeometry);
+        numErrors++;
+      }
+      var mltProperties = mltFeature.properties();
+      var mvtProperties = mvtFeature.properties();
+      for (var mvtProperty : mvtProperties.entrySet()) {
+        var mvtPropertyKey = mvtProperty.getKey();
+        // TODO: remove the below special case once this bug is fixed:
+        // https://github.com/maplibre/maplibre-tile-spec/issues/181
+        if (mvtPropertyKey.equals("id")) {
+          continue;
+        }
+        var mltProperty = mltProperties.get(mvtPropertyKey);
+        if (mltProperty == null) {
+          // System.out.println(mvtFeature.id() + " mlt prop missing for " + mvtProperty.getKey());
+          // System.out.println("  mvtProperties: " + mvtProperties);
+          // System.out.println("  mltProperties: " + mltProperties);
+          numErrors++;
+        } else if (!mltProperty.equals(mvtProperty.getValue())) {
+          // System.out.println(
+          //     "Failure comparing property "
+          //         + mvtProperty.getKey()
+          //         + " for feature: "
+          //         + mvtFeature.id());
+          // System.out.println("    mvtProperty: " + mvtProperty.getValue());
+          // System.out.println("    mltProperty: " + mltProperty);
```

Comment:
Left in intentionally since these debug prints can be handy when trying to understand a failure. But commented so the current tests will not be extremely noisy (rather the number of errors is asserted on to keep things calm and readable)